File: classlog-frontend/src/app/model/entities/question-dto.ts
import {FileDto} from "./file-dto";
import {QuestionType} from "./question-type-dto";

export interface QuestionDto {
  questionId: number;
  questionType: QuestionType;
  editedAt: Date;
  points: number;
  content: string;
  fileId: FileDto | null;
}

File: classlog-backend/src/main/java/com/example/classlog/dto/QuestionDto.java
package com.example.classlog.dto;

import com.example.classlog.entity.QuestionType;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class QuestionDto {

  private Long questionId;
  private QuestionType questionType;
  private LocalDateTime editedAt;
  private Integer points;
  private String content;
  private FileDto file;
}

